python/sanlock.c

```diff
@@ -128,7 +128,7 @@ __parse_resource(PyObject *obj, struct sanlk_resource **res_ret)
         if (offset == NULL) {
             res->disks[i].offset = 0;
         } else {
-            res->disks[i].offset = PyInt_AsLong(offset);
+            res->disks[i].offset = PyLong_AsLong(offset);
```

Comment:
probably should be PyLong_AsUnsignedLong(offset)  since "unsigned long long" is not strongly typed on gcc systems and depends on actual register size, as can be seen in /usr/include/bits/types.h:
```
#if __WORDSIZE == 64
typedef signed long int __int64_t;
typedef unsigned long int __uint64_t;
#else
__extension__ typedef signed long long int __int64_t;
__extension__ typedef unsigned long long int __uint64_t;
#endif
```
Also discussed in https://stackoverflow.com/questions/32198368/unsigned-long-long-conflict-with-uint64-t

Reply:
I'm not sure what do you mean by "not strongly typed".

Python does not give you a way to specify uint64_t, only PY_LONG_LONG. I think this is the correct type if you want to convert from/to [u]int64_t, ensuring that your code will be portable.
See http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf

If you are not sure about that, why not ask on python mailing?

Reply:
havent got to the mailing yet, but is seems that PyLong_AsUnsignedLongLong(offset) fails and returns ((ULL)(-1)) when offset object is not explicitly PyLong in python2. PyLong_AsUnsignedLong(offset) doesn't fail when this happens. 
Seems the py2 impl has fallback in PyLong_AsUnsignedLong(x) when x is not PyLong but a PyInt [1],  which PyLong_AsUnsignedLongLong does not possess[2].
[1] https://github.com/certik/python-2.7/blob/c360290c3c9e55fbd79d6ceacdfc7cd4f393c1eb/Objects/longobject.c#L397
[2] https://github.com/certik/python-2.7/blob/c360290c3c9e55fbd79d6ceacdfc7cd4f393c1eb/Objects/longobject.c#L973

Reply:
So we have 2 options:
- add a helpers that accept both PyInt and PyLong, similar to is_py_integer.
- support only PyInt input on python 2, since we never supported PyLong.

Since our goal is to have python 3 version, and making python 2 version better is non-goal, I think the second option is best.

To keep python 2 version consistent, we can return also PyInt from sanlcok on py2, keeping current behavior.

Reply:
me think that we'll need helper in both cases 
as we need to call PyInt_XXXX in py2 and PyLong_XXXX in py3.

Reply:
@aabawer, I have alternative patch for parsing intput:
https://github.com/nirs/sanlock/commit/d3b7f20589dcdfc895ab4d44d404d12fd7e978bf

If we parse input with "K", we have no problem to accept long values, so there is no problem to return long values.

What do you think?